File: runtime/src/de/fatox/meta/MetaLogger.java
package de.fatox.meta;

import de.fatox.meta.api.Logger;

public class MetaLogger implements Logger {
    public LogLevel logLevel = LogLevel.TRACE;

    @Override
    public void trace(String tag, String text) {
        if (logLevel.ordinal() <= LogLevel.TRACE.ordinal()) {
            System.out.println("[trace] - " + tag + " : " + text);
        }
    }

    @Override
    public void info(String tag, String text) {
        if (logLevel.ordinal() <= LogLevel.INFO.ordinal()) {
            System.out.println("[info] - " + tag + " : " + text);
        }
    }

    @Override
    public void debug(String tag, String text) {
        if (logLevel.ordinal() <= LogLevel.DEBUG.ordinal()) {
            System.out.println("[debug] - " + tag + " : " + text);
        }
    }

    @Override
    public void warning(String tag, String text) {
        if (logLevel.ordinal() <= LogLevel.WARNING.ordinal()) {
            System.out.println("[warning] - " + tag + " : " + text);
        }
    }

    @Override
    public void error(String tag, String text) {
        if(logLevel.ordinal() <= LogLevel.ERROR.ordinal()) {
            System.err.println("[error] - " + tag + " : " + text);
        }
    }
}

File: runtime/src/de/fatox/meta/MetaLogger.kt
package de.fatox.meta

import de.fatox.meta.api.Logger

class MetaLogger : Logger {
    var logLevel: Logger.LogLevel = Logger.LogLevel.TRACE

    override fun trace(tag: String, text: String) {
        if (logLevel.ordinal <= Logger.LogLevel.TRACE.ordinal) {
            println("[trace] - $tag : $text")
        }
    }

    override fun info(tag: String, text: String) {
        if (logLevel.ordinal <= Logger.LogLevel.INFO.ordinal) {
            println("[info] - $tag : $text")
        }
    }

    override fun debug(tag: String, text: String) {
        if (logLevel.ordinal <= Logger.LogLevel.DEBUG.ordinal) {
            println("[debug] - $tag : $text")
        }
    }

    override fun warning(tag: String, text: String) {
        if (logLevel.ordinal <= Logger.LogLevel.WARNING.ordinal) {
            println("[warning] - $tag : $text")
        }
    }

    override fun error(tag: String, text: String) {
        if (logLevel.ordinal <= Logger.LogLevel.ERROR.ordinal) {
            System.err.println("[error] - $tag : $text")
        }
    }
}
